source/fe/mapping_q_generic.cc

```diff
@@ -3877,8 +3833,16 @@ compute_mapping_support_points(const typename Triangulation<dim,spacedim>::cell_
         // manifold, otherwise compute them from the points around it
         if (dim != spacedim)
           add_quad_support_points(cell, a);
+        else if (dynamic_cast<const Boundary<dim,spacedim> *>(&cell->get_manifold()) == NULL)
+          {
+            std::vector<Point<spacedim> > quad_points (Utilities::fixed_power<dim>(polynomial_degree-1));
+            get_intermediate_points_on_object(cell->get_manifold(), line_support_points, cell, quad_points);
+            for (unsigned int i=0; i<quad_points.size(); ++i)
+              a.push_back(quad_points[i]);
+          }
         else
-          add_weighted_interior_points (support_point_weights_on_quad, a);
+          add_weighted_interior_points(support_point_weights_on_quad, a);
```

Comment:
@luca-heltai you suggested to let the `else` branch also use the `else if` if I understood correctly (now hidden). Unfortunately, I could not remove this code (-> test suite). If a `Boundary` object is attached to the grid, it runs into:
```
#4  0x00007ffff5a637ec in dealii::StraightBoundary<2, 2>::get_intermediate_points_on_quad (this=<optimized out>)
    at /home/kronbichler/deal/deal.II/source/grid/tria_boundary.cc:346
#5  0x00007ffff418f363 in dealii::(anonymous namespace)::get_intermediate_points_on_object<2, 2, dealii::TriaIterator<dealii::CellAccessor<2, 2> > > (manifold=..., line_support_points=..., iter=..., 
    points=std::vector of length 9, capacity 9 = {...})
    at /home/kronbichler/deal/deal.II/source/fe/mapping_q_generic.cc:3595
#6  0x00007ffff418f59f in dealii::MappingQGeneric<2, 2>::compute_mapping_support_points (this=0x54cb40, cell=...)
    at /home/kronbichler/deal/deal.II/source/fe/mapping_q_generic.cc:3839
```
i.e., it tries to find intermediate points on a quad in 2D which is not implemented. Unless there is something that I'm overlooking right now, I would keep things as they are.